--- a/main/java/net/minecraft/client/gui/screens/DisconnectedScreen.java
+++ b/main/java/net/minecraft/client/gui/screens/DisconnectedScreen.java
@@ -4,7 +4,9 @@
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.MultiLineLabel;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 import net.minecraft.network.chat.TranslatableComponent;
+import si.f5.pa_union.reconnect.Reconnect;
 
 public class DisconnectedScreen extends Screen
 {
@@ -29,10 +31,14 @@
     {
         this.message = MultiLineLabel.create(this.font, this.reason, this.width - 50);
         this.textHeight = this.message.getLineCount() * 9;
-        this.addRenderableWidget(new Button(this.width / 2 - 100, Math.min(this.height / 2 + this.textHeight / 2 + 9, this.height - 30), 200, 20, new TranslatableComponent("gui.toMenu"), (p_96002_) ->
+        this.addRenderableWidget(new Button(this.width / 2 - 100, Math.min(this.height / 2 + this.textHeight / 2 + 9, this.height - 30) + 25, 200, 20, new TranslatableComponent("gui.toMenu"), (p_96002_) ->
         {
             this.minecraft.setScreen(this.parent);
         }));
+        this.addRenderableWidget(new Button(this.width / 2 - 100, Math.min(this.height / 2 + this.textHeight / 2 + 9, this.height - 30), 200, 20, new TextComponent("再接続"), (p_96002_) ->
+        {
+            Reconnect.reconnect();
+        }));
     }
 
     public void render(PoseStack pMatrixStack, int pMouseX, int pMouseY, float pPartialTicks)
