--- a/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -34,6 +34,7 @@
 import net.minecraft.core.RegistryAccess;
 import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
 import net.minecraft.network.chat.TranslatableComponent;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.MinecraftServer;
@@ -45,6 +46,7 @@
 import net.optifine.reflect.ReflectorForge;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import si.f5.pa_union.update.UpdateScreen;
 
 public class TitleScreen extends Screen
 {
@@ -211,13 +213,15 @@
             Screen screen = (Screen)(this.minecraft.options.skipMultiplayerWarning ? new JoinMultiplayerScreen(this) : new SafetyScreen(this));
             this.minecraft.setScreen(screen);
         }, button$ontooltip))).active = flag;
-        boolean flag1 = Reflector.ModListScreen_Constructor.exists();
-        int i = flag1 ? this.width / 2 + 2 : this.width / 2 - 100;
-        int j = flag1 ? 98 : 200;
+        int i = this.width / 2 + 2;
+        int j = 98;
         (this.addRenderableWidget(new Button(i, pY + pRowHeight * 2, j, 20, new TranslatableComponent("menu.online"), (p_96775_1_) ->
         {
             this.realmsButtonClicked();
         }, button$ontooltip))).active = flag;
+        (this.addRenderableWidget(new Button(this.width / 2 - 100, pY + pRowHeight * 2, 98, 20, new TextComponent("更新情報"), (p_96788_) -> {
+            this.minecraft.setScreen(new UpdateScreen());
+        }, button$ontooltip))).active = flag;
     }
 
     private void createDemoMenuOptions(int pY, int pRowHeight)
