--- a/main/java/net/minecraft/client/gui/components/PlayerTabOverlay.java
+++ b/main/java/net/minecraft/client/gui/components/PlayerTabOverlay.java
@@ -27,6 +27,7 @@
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard;
 import net.minecraft.world.scores.criteria.ObjectiveCriteria;
+import si.f5.pa_union.KurarClient;
 
 public class PlayerTabOverlay extends GuiComponent
 {
@@ -62,8 +63,12 @@
 
     private Component decorateName(PlayerInfo p_94552_, MutableComponent p_94553_)
     {
+        if (KurarClient.getInstance().canSeeTrueTabList()) {
         return p_94552_.getGameMode() == GameType.SPECTATOR ? p_94553_.withStyle(ChatFormatting.ITALIC) : p_94553_;
+        } else {
+            return p_94553_;
     }
+    }
 
     public void setVisible(boolean pVisible)
     {
@@ -203,19 +208,30 @@
                     k2 += 9;
                 }
 
-                this.minecraft.font.drawShadow(p_94545_, this.getNameForDisplay(playerinfo1), (float)k2, (float)l2, playerinfo1.getGameMode() == GameType.SPECTATOR ? -1862270977 : -1);
+                if (KurarClient.getInstance().canSeeTrueTabList()) {
+                    this.minecraft.font.drawShadow(p_94545_, this.getNameForDisplay(playerinfo1), (float) k2, (float) l2, playerinfo1.getGameMode() == GameType.SPECTATOR ? -1862270977 : -1);
 
-                if (p_94548_ != null && playerinfo1.getGameMode() != GameType.SPECTATOR)
-                {
+                    if (p_94548_ != null && playerinfo1.getGameMode() != GameType.SPECTATOR) {
                     int l5 = k2 + i + 1;
                     int i6 = l5 + l;
 
-                    if (i6 - l5 > 5)
-                    {
+                        if (i6 - l5 > 5) {
                         this.renderTablistScore(p_94548_, l2, gameprofile.getName(), l5, i6, playerinfo1, p_94545_);
                     }
                 }
+                } else {
+                    this.minecraft.font.drawShadow(p_94545_, this.getNameForDisplay(playerinfo1), (float) k2, (float) l2, -1);
 
+                    if (p_94548_ != null) {
+                        int l5 = k2 + i + 1;
+                        int i6 = l5 + l;
+
+                        if (i6 - l5 > 5) {
+                            this.renderTablistScore(p_94548_, l2, gameprofile.getName(), l5, i6, playerinfo1, p_94545_);
+                        }
+                    }
+                }
+
                 this.renderPingIcon(p_94545_, i1, k2 - (flag ? 9 : 0), l2, playerinfo1);
             }
         }
@@ -390,7 +406,11 @@
         {
             PlayerTeam playerteam = p_94564_.getTeam();
             PlayerTeam playerteam1 = p_94565_.getTeam();
+            if (KurarClient.getInstance().canSeeTrueTabList()) {
             return ComparisonChain.start().compareTrueFirst(p_94564_.getGameMode() != GameType.SPECTATOR, p_94565_.getGameMode() != GameType.SPECTATOR).compare(playerteam != null ? playerteam.getName() : "", playerteam1 != null ? playerteam1.getName() : "").compare(p_94564_.getProfile().getName(), p_94565_.getProfile().getName(), String::compareToIgnoreCase).result();
+            } else {
+                return ComparisonChain.start().compare(playerteam != null ? playerteam.getName() : "", playerteam1 != null ? playerteam1.getName() : "").compare(p_94564_.getProfile().getName(), p_94565_.getProfile().getName(), String::compareToIgnoreCase).result();
         }
     }
 }
+}
